### ISSUE TITLE
Add DisapproveRating and extend AggregateRating

### ISSUE THREAD

**Issue by thadguidry** [schemaorg/schemaorg#2148]
In contrast to #1293 with [EndorsementRating](https://schema.org/EndorsementRating).
Introduce a new type called "DisapproveRating".

```json
        "review": [{
            "@type": "Review",
            "name": "The Ratings Game",
            "reviewRating": [{
                "@type": "Rating",
                "bestRating": "5",
                "ratingValue": "1",
                "worstRating": "1"
            },{
                "@type": "DisapproveRating",
                "name": "1 rotten tomato",
                "ratingValue": "1"
            }]
```

Also many sites represent explicit **counts** of Endorsements and Disapprovals on an issue or Thing by their members with units such as Thumbs Up/Thumbs Down or Fresh/Rotten, for example.

![capture](https://user-images.githubusercontent.com/986438/53016619-a04dff80-3413-11e9-9954-d699d7211059.PNG)

We should add a way to hold **aggregates** of those Endorsements or Dispprovals by extending AggregateRating with new properties.

The result would look something like:

```json
{
  "@context": "http://schema.org",
  "@type": "Movie",
  "name": "The Ratings Game",
  "actor": "Danny Devito, Rhea Perlman, Gerrit Graham, Bernadette Birkett",
  "director": "Danny Devito",
  "datePublished": "1984",
  "aggregateRating": {
    "@type": "AggregateRating",
    "reviewCount": "318",
    "ratingValue": "6.3",
    "aggregateEndorsements": "219",
    "aggregateDisapprovals": "99"
  }
}
```

What's missing is a way to capture the scale, in the example "10", of the average rating value "6.3".  Or did I miss something somewhere ?

**Comment by vholland** [comment]
Shouldn't https://schema.org/interactionStatistic work for this use case?

**Comment by thadguidry** [comment]
@vholland It could I guess.  I'm not sure what that looks like because we really are lacking some examples of this method...here's a quick stab at it however... It seems that AssessAction has the right definition for this use case however, if I'm not mistaken ?

```json
{
  "@context": "http://schema.org",
  "interactionStatistic": {
    "@type": "http://schema.org/InteractionCounter",
    "name": "Thumbs Up",
    "interactionType": {
      "@type": "http://schema.org/AssessAction",
      "description": "AssessAction - The act of forming one's opinion, reaction or sentiment.",
      "name": "Thumbs Up",
      "actionStatus": {
        "@type": "https://schema.org/CompletedActionStatus"
      },
      "userInteractionCount": "219"
    }
  }
}
```

**Comment by vholland** [comment]
I was thinking something similar with LikeAction or DislikeAction:
```
{
  "@context": "http://schema.org",
  "interactionStatistic": {
    "@type": "http://schema.org/InteractionCounter",
    "name": "Thumbs Up count",
    "interactionType": {
      "@type": "http://schema.org/LikeAction"
      },
      "userInteractionCount": "219"
    }
  }
}
```

**Comment by thadguidry** [comment]
AH! We have LikeAction and DislikeAction...but only EndorseAction.
Yes, I think your framing is correct on how to handle the individual .
We should get this into our examples, or wiki, once these remaining 2 questions are resolved: 

1. So now for the AggregateRating case, you think we can avoid extending that and just multiply with interactionStatistic?
2. How to deal with the max range (scale) for Average Rating 10 in my Rotten Tomatoes example snippet ?

```json
{
  "@context": "http://schema.org",
  "@type": "Movie",
  "name": "The Ratings Game",
  "actor": "Danny Devito, Rhea Perlman, Gerrit Graham, Bernadette Birkett",
  "director": "Danny Devito",
  "datePublished": "1984",
  "aggregateRating": {
    "@type": "AggregateRating",
    "reviewCount": "318",
    "ratingValue": "6.3"
  },
  "interactionStatistic": {
    "@type": "http://schema.org/InteractionCounter",
    "name": "Thumbs Up count",
    "interactionType": {
      "@type": "http://schema.org/LikeAction"
      },
      "userInteractionCount": "219"
  },
  "interactionStatistic": {
    "@type": "http://schema.org/InteractionCounter",
    "name": "Thumbs Down count",
    "interactionType": {
      "@type": "http://schema.org/DislikeAction"
      },
      "userInteractionCount": "99"
    }
}
```

**Comment by MichaelAndrews-RM** [comment]
@thadguidry We also have a [downvoteCount](https://schema.org/downvoteCount) -- could that be used?

**Comment by thadguidry** [comment]
@MichaelAndrews-RM Yes and No.  The downvoteCount was created to capture simply what we would call a "-1" on a question, answer, or comment.  Examples that you commonly see on sites such as StackOverflow.  I think a "-1" is closer to the concept of https://schema.org/DisagreeAction  And in fact when you have a DisagreeAction, they could be captured in aggregate with userInteractionCount.

Sentiment is what is captured with LikeAction and DislikeAction, and then the counts of those as @vholland shows can be captured with userInteractionCount...probably for ANY Action that involves users.

The downvoteCount is currently range limited to only Questions and Comments.
The interactionStatistic can be used on any CreativeWork or subtype.

So for Questions and Comments, you could capture with downvoteCount OR with lots of extra structure with interactionStatistic and userInteractionCount... and probably little value for capturing that extra structure.

**Comment by MichaelAndrews-RM** [comment]
@thadguidry Understand that currently range of upvoteCount/downvoteCount is limited to questions and comments.  I think they should be promoted to cover any Thing.  Would that break anything currently in place if that were done?  

Compared to ratings, which are generally on a scale, voting is generally binary (yes/no).  Basically we are looking for an equivalent to thumbs down or thumbs up, which appear next to videos (e.g.,  YouTube), customer product reviews (what that helpful? yes/no), and many other things online.  While most will be Creative Works, they could be applied to places, organizations, or people as well.  In some cases, one may want to associate a specific question/statement relating to these entities, so it is clear precisely what people are expressing.  Many people express frustration that "like" buttons don't really capture the intent of what they are trying to communicate.  And the vocabulary doesn't seem to have an entity covering a poll that might have a yes/no response, is that correct?

You raise a good point about the alignment with actions.  While it might be similar to DislikeAction (a personal preference), it could also be associated with the VoteAction (a verdict).  Votes don't have to be binary, but many people consider any expression that involves a competitive or comparative count as voting.  Liking a photo is not a vote, but choosing whether or not the photo is worthy would be.  I think of people expressing views on Rotten Tomatoes as voting, because they are rendering a judgment.

**Comment by thadguidry** [comment]
@MichaelAndrews-RM disagree on the point that "voting" can be applied directly to places, orgs, people.  Voting happens on an idea, comment or question about places, orgs, people.  Ex.  Would you vote for Trump in the next election?  or  I think we should create another planet because this one is tired looking now, do you agree?  Instead of "Trump - Y/N" or "Mars - Y/N"  Its more likely you will see Likes Dislikes and general sentiment about Things themselves and Voting concerning Questions, Comments about those Things.

**Comment by MichaelAndrews-RM** [comment]
Hi Thad, I'm just trying to parse the distinction between liking and voting, which can get a bit blurry.  I think voting is a better option when people are expressing a decision.  Opinion polls measure favorable and unfavorable ratings, while election polls decide outcomes.   For a movie, I may have an opinion that could change, or I may have made a decision about it that won't change.  You don't have to agree with that characterization, but I do feel the documentation currently doesn't offer much guidance.

Here's my thinking: when people vote, they do not necessarily have to have both an explicit yes and no choice.  They may only get a yes choice for each option, which would be reflected in the upvoteCount. They get more than one option if they can't downvote.    

I do think people can vote about tangible things, like people and locations.  Imagine voting for candidates for a committee of an organization.  Those are people.  Right now the vocabulary doesn't allow for the an upvoteCount of people running in an election.  But the vocabulary could potentially do that.  One idea would be to describe the Person in a role as a candidate.  The vocabulary already has a [candidate](https://schema.org/candidate) property that's associated with the VoteAction.  Cities and organizations can also be candidates for different roles: as host of an event, for example.  

Perhaps other kinds of things could also have roles that could be voted on.  More generally, products might have different AggregateRatings according to quality, value, ease-of-use, etc.  The examples for role in the documentation don't suggest that sort of use, but it seems to me a useful possibility.

**Comment by thadguidry** [comment]
They don't "vote" for people... they "vote for people to hold an office or title".  It's not voting for "Thad".  It's saying "I vote for Thad to be mayor of Schema.org City".  Its expanded, but we often just shorten and concise things with "I voted for him", rather than "I voted for **Thad** to be **mayor** of **Schema.org** in **2017** on an **absentee ballot**".  In other words...there's a lot more to say about Actions in general, and the Roles that People play.  I see only gaps in documentation and tons of missing examples like we are talking about in this issue now.  But most of the high-level abstracts are coming together nicely in Schema.org for the most part.

Anyways...I don't like ranting... I prefer proving out gaps with close-up examples rather than long discussions about what could be done or not or how.

For now, I am ok with using interactionStatistic and it satisfies the need, although not the max range (scale).
